daras_ai_v2/base.py

```diff
@@ -514,23 +520,35 @@ def _render_published_run_save_buttons(
             )
             publish_modal = gui.Modal("Publish to", key="publish-modal")
             if pressed_save:
-                publish_modal.open()
+                if self.request.user and not self.request.user.is_anonymous:
+                    publish_modal.open()
+                else:
+                    self._save_for_anonymous_user(current_run=current_run)
             if publish_modal.is_open():
                 with publish_modal.container(style={"minWidth": "min(500px, 100vw)"}):
                     self._render_publish_modal(
                         current_run=current_run,
                         published_run=published_run,
                         modal=publish_modal,
-                        is_update_mode=can_edit,
                     )
 
+    def _save_for_anonymous_user(self, current_run: SavedRun):
+        if self._has_request_changed():
```

Comment:
I believe what @SeanBlagsvedt asked for is that the run not be actually charged under anon but under the logged in user?